### PR TITLE
[REVIEW] Fix `categories` duplication in `dask_cudf`

### PR DESCRIPTION
**Pull request by galipremsagar**
Fixes: #2643 

1. We seem to go down to `__getattr__` path which internally calls `_property_map` which will be combining the categories fetched from all partitions, instead now we have removed all Accessors from Dask_cudf and will rely on upstream `dask` implementations.

2. Added test coverage for removed accessors.

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
cc @quasiben would be good for you to review as well

**Comment by galipremsagar** [comment]
> How feasible would it be to remove the accessor classes altogether from dask-cudf and rely on upstream for the implementations? How much would need to change upstream?

@kkraus14 I just removed both `CategoricalAccessor` and `DatetimeAccessor` and all the test coverage seems to be working and did sanity check on accessing their attributes and they also seem to work fine. Do you want it to be scoped into this PR? or a separate PR?

**Comment by kkraus14** [comment]
> > How feasible would it be to remove the accessor classes altogether from dask-cudf and rely on upstream for the implementations? How much would need to change upstream?
> 
> @kkraus14 I just removed both `CategoricalAccessor` and `DatetimeAccessor` in `dask-cudf` and all the test coverage seems to be working and did sanity check on accessing their attributes and they also seem to work fine. Do you want it to be scoped into this PR? or a separate PR?

Did you confirm they aren't being converted to host objects erroneously? I.E. want to make sure we're not silently getting numpy arrays or Pandas Series objects. I think it would be better for this PR to just remove those accessors entirely if it's feasible 😄.

**Comment by galipremsagar** [comment]
> > > How feasible would it be to remove the accessor classes altogether from dask-cudf and rely on upstream for the implementations? How much would need to change upstream?
> > 
> > 
> > @kkraus14 I just removed both `CategoricalAccessor` and `DatetimeAccessor` in `dask-cudf` and all the test coverage seems to be working and did sanity check on accessing their attributes and they also seem to work fine. Do you want it to be scoped into this PR? or a separate PR?
> 
> Did you confirm they aren't being converted to host objects erroneously? I.E. want to make sure we're not silently getting numpy arrays or Pandas Series objects. I think it would be better for this PR to just remove those accessors entirely if it's feasible 😄.

Yes, I did check some attribute types, they were all device objects(`cudf.DataFrame`, & `cudf.Series`). But will definitely double-check all of the types and remove those accessors in this PR.

**Comment by galipremsagar** [comment]
> But will definitely double-check all of the types and remove those accessors in this PR.

@kkraus14 @quasiben, Changes are ready.

File: python/dask_cudf/dask_cudf/tests/test_accessor.py [hunk 1]
```diff
@@ -24,12 +24,12 @@ def data_dt_2():
...
 
 @pytest.mark.parametrize("data", [data_dt_2()])
-@pytest.mark.xfail(raises=AttributeError)
 def test_datetime_accessor_initialization(data):
     pdsr = pd.Series(data.copy())
     sr = Series(pdsr)
     dsr = dgd.from_cudf(sr, npartitions=5)
-    dsr.dt
+    with pytest.raises(AttributeError):
+        dsr.dt
```
Comment (kkraus14) [inline python/dask_cudf/dask_cudf/tests/test_accessor.py:32]: Why do we raise an AttributeError here? Does the datetime accessor not work as intended?

File: python/dask_cudf/dask_cudf/tests/test_accessor.py [hunk 2]
```diff
@@ -76,15 +89,21 @@ def data_cat_3():
...
     dsr.cat
 
 
-@pytest.mark.xfail(reason="")
+@pytest.mark.parametrize("data", [data_cat_2()])
+def test_categorical_accessor_initialization2(data):
+    sr = Series(data.copy())
+    dsr = dgd.from_cudf(sr, npartitions=5)
+    with pytest.raises(AttributeError):
+        dsr.cat
```
Comment (kkraus14) [inline python/dask_cudf/dask_cudf/tests/test_accessor.py:104]: here as well

Comment (galipremsagar) [reply python/dask_cudf/dask_cudf/tests/test_accessor.py:104]: The data in `dsr` is of not Datetime type. It's a Int Series here.

File: python/dask_cudf/dask_cudf/tests/test_accessor.py (hunk 1 above)
Comment (galipremsagar) [reply python/dask_cudf/dask_cudf/tests/test_accessor.py:32]: The data in `dsr` is of not Datetime type. It's a float Series.

**Pull request merged**